src/ui/views/Setting/Account/index.tsx

```diff
@@ -144,10 +156,69 @@ const AccountSettings = () => {
     await updatePreference(!modeAnonymous);
   }, [updatePreference, modeAnonymous]);
 
+  const handleOpenRemoveModal = () => {
+    setRemoveError('');
+    setShowRemoveConfirmModal(true);
+  };
+
+  const handleCloseRemoveModal = () => {
+    setShowRemoveConfirmModal(false);
+    setRemoveError('');
+  };
+
+  const handleConfirmRemove = async (password: string) => {
+    if (!activeAccountDetails) {
+      console.error('Cannot remove profile: No active account details found.');
+      setRemoveError('Account details not loaded for removal.');
+      return;
+    }
+
+    setIsRemoving(true);
+    setRemoveError('');
+
+    try {
+      // Get the current profile ID
+      const userInfo = await wallet.getUserInfo(false);
+      const profileId = userInfo.id;
+
+      if (!profileId) {
+        throw new Error('Could not determine profile ID');
+      }
```

Comment:
Use getCurrentProfileId() - note you need to import it too

```suggestion
    const profileId = await getCurrentProfileId();

```